src/generate-oas.ts

```diff
@@ -0,0 +1,197 @@
+  }
+  return name.replace(/\s+/g, "").replace("OAS", "");
+}
+
+/**
+ * Helper that reads api details from exchange.json
+ */
+export function getAPIDetailsFromExchange(directory: string): ApiSpecDetail {
+  const exchangePath = path.join(directory, "exchange.json");
+  if (fs.existsSync(exchangePath)) {
+    const exchangeConfig = fs.readJSONSync(
+      exchangePath
+    ) as download.ExchangeConfig;
+
+    // Special handling for shopper-baskets-v2
+    if (
+      exchangeConfig.assetId === "shopper-baskets-oas" &&
+      exchangeConfig.apiVersion === "v2"
+    ) {
+      return {
+        filepath: path.join(directory, exchangeConfig.main),
+        filename: exchangeConfig.main,
+        directoryName: "ShopperBasketsV2",
+        name: "Shopper Baskets V2 OAS",
+        apiName: "ShopperBasketsV2",
+      };
+    }
+
+    return {
+      filepath: path.join(directory, exchangeConfig.main),
+      filename: exchangeConfig.main,
+      directoryName: kebabToCamelCase(
+        exchangeConfig.assetId.replace("-oas", "")
+      ),
+      name: exchangeConfig.name,
+      apiName: resolveApiName(exchangeConfig.name),
+    };
+  }
+  throw new Error(`Exchange file does not exist for ${directory}`);
+}
+
+/**
+ * Invokes openapi-generator via raml-toolkit to generate SDKs
+ */
+export function generateSDKs(apiSpecDetail: ApiSpecDetail): void {
+  const { filepath, name, directoryName } = apiSpecDetail;
+  const fileIsOASspec = filepath.includes("oas");
+  if (fs.statSync(filepath).isFile() && fileIsOASspec) {
+    try {
+      console.log(`Generating SDK for ${name}`);
+      const outputDir = `${TARGET_DIRECTORY}/${directoryName}`;
+      generateFromOas.generateFromOas({
+        inputSpec: `${filepath}`,
+        outputDir: `${outputDir}`,
+        templateDir: `${TEMPLATE_DIRECTORY}`,
+        flags: `--reserved-words-mappings delete=delete`,
+      });
+    } catch (error) {
+      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+      console.error(`Error generating SDK for ${name}: ${error}`);
+    }
+  }
+}
+
+/**
+ * Generates the top level index file
+ */
+export function generateIndex(context: {
+  children: ApiSpecDetail[] | { name: string; apiName: string }[];
+}): void {
+  const indexTemplate = fs.readFileSync(INDEX_TEMPLATE_LOCATION, "utf8");
+  const generatedIndex = Handlebars.compile(indexTemplate)(context);
+  fs.writeFileSync(`${TARGET_DIRECTORY}/index.ts`, generatedIndex);
+}
+
+function getAllDirectories(basePath: string, relativePath = ""): string[] {
+  const fullPath = path.join(basePath, relativePath);
+  const directories: string[] = [];
+
+  try {
+    const items = fs.readdirSync(fullPath);
+
+    for (const item of items) {
+      const itemPath = path.join(fullPath, item);
+      const relativeItemPath = relativePath
+        ? path.join(relativePath, item)
+        : item;
+
+      if (fs.lstatSync(itemPath).isDirectory()) {
+        directories.push(relativeItemPath);
```

Comment:
in case of baskets-oas, this seem to add three dirs to the  `directories` array. `baskets-oas, shopper-baskets-oas-1.8.19, shopper-baskets-oas-2.0.10`. Is that correct? if so, we should only add the dirs that has the actual spec. May be check for exchange.json in the dir

Reply:
Good suggestion! I've added this change.